pkg/infrastructure/services/dashboards.go

```diff
@@ -0,0 +1,147 @@
+	if err != nil {
+		return nil, err
+	}
+	if !available {
+		log.Debugf("Deployment not available yet for KogitoService %s ", instance.GetName())
+		return nil, nil
+	}
+
+	svcURL := infrastructure.CreateKogitoServiceURI(instance)
+	dashboardNames, err := fetchGrafanaDashboardNamesForURL(svcURL)
+	if err != nil {
+		return nil, err
+	}
+
+	return fetchDashboards(svcURL, dashboardNames)
+}
+
+func fetchGrafanaDashboardNamesForURL(serverURL string) ([]string, error) {
+	dashboardsURL := fmt.Sprintf("%s%s%s", serverURL, dashboardsPath, "list.json")
+	resp, err := http.Get(dashboardsURL)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode == http.StatusNotFound {
+		return nil, nil
+	}
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("Received NOT expected status code %d while making GET request to %s ", resp.StatusCode, dashboardsURL)
+	}
+	var dashboardNames []string
+	if err := json.NewDecoder(resp.Body).Decode(&dashboardNames); err != nil {
+		return nil, fmt.Errorf("Failed to decode response from %s into dashboard names", dashboardsURL)
+	}
+
+	return dashboardNames, nil
+}
+
+func fetchDashboards(serverURL string, dashboardNames []string) ([]GrafanaDashboard, error) {
+	var dashboards []GrafanaDashboard
+	for _, name := range dashboardNames {
+		dashboardURL := fmt.Sprintf("%s%s%s", serverURL, dashboardsPath, name)
+		resp, err := http.Get(dashboardURL)
+		if err != nil {
+			return nil, err
+		}
+		defer resp.Body.Close()
+		if resp.StatusCode == http.StatusNotFound {
+			return nil, nil
+		}
+		if resp.StatusCode != http.StatusOK {
+			return nil, fmt.Errorf("Received NOT expected status code %d while making GET request to %s ", resp.StatusCode, dashboardURL)
+		}
+		bodyBytes, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			log.Fatal(err)
+		}
+		dashboards = append(dashboards, GrafanaDashboard{Name: name, RawJSONDashboard: string(bodyBytes)})
+	}
+	return dashboards, nil
+}
+
+func configureGrafanaDashboards(client *client.Client, kogitoService v1alpha1.KogitoService, scheme *runtime.Scheme, namespace string) (time.Duration, error) {
+	dashboards, err := fetchGrafanaDashboards(client, kogitoService)
+	if err != nil {
+		return reconciliationPeriodAfterDashboardsError, err
+	}
+
+	reconcileAfter, err := deployGrafanaDashboards(dashboards, client, kogitoService, scheme, namespace)
+
+	return reconcileAfter, err
+}
+
+func deployGrafanaDashboards(dashboards []GrafanaDashboard, cli *client.Client, kogitoService v1alpha1.KogitoService, scheme *runtime.Scheme, namespace string) (time.Duration, error) {
+	for _, dashboard := range dashboards {
+		resourceName := strings.ReplaceAll(strings.ToLower(dashboard.Name), ".json", "")
+		dashboardDefinition := &grafanav1.GrafanaDashboard{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      resourceName,
+				Namespace: namespace,
+				Labels: map[string]string{
+					framework.LabelAppKey: GrafanaDashboardAppName,
```

Comment:
Could we provide actual kogitoService name in place of hardcoded `grafana` keyword?

Reply:
Hi @vaibhavjainwiz , if I use the kogito service name (or any other string), I get the following error in the grafana operator `":"dashboard trusty-demo/operational-dashboard-loaneligibility found but selectors do not match"}` . 
Only with `grafana` it works. Any thoughts?

Reply:
I believe it's a requirement for the Grafana operator. We might need to take a look at their docs. Plus, our label in here won't hurt like `kogito.kie.org/service: serviceName`.

Reply:
I suppose it is because the default Grafana CR from OCP contains:
```
  dashboardLabelSelector:
    - matchExpressions:
        - key: app
          operator: In
          values:
            - grafana
```

Reply:
IMHO we can put there the usual application name, possibly adding a note in docs for user to set proper selector in Grafana CR

Reply:
Thanks for the info, @sutaakar 
That's the proper way of handling this, imo.

Reply:
Hi @sutaakar @ricardozanini , I've changed the label name to `kogitoService.GetName()`, thanks for the tip!